new resource `azurerm_stack_hci_logical_network` (#26473)
hashicorp/terraform-provider-azurerm: website/docs/r/stack_hci_logical_network.html.markdown

  virtual_switch_name = "ConvergedSwitch(managementcompute)"
  dns_servers         = ["10.0.0.7", "10.0.0.8"]

  subnet {
    ip_allocation_method = "Static"
    address_prefix       = "10.0.0.0/24"
    route {
      name                = "example-route"
      address_prefix      = "0.0.0.0/0"
      next_hop_ip_address = "10.0.20.1"
    }
    vlan_id = 123
  }

  tags = {
    foo = "bar"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Azure Stack HCI Logical Network. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

* `custom_location_id` - (Required) The ID of Custom Location where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

> * `virtual_switch_name` - (Required) The name of the virtual switch on the cluster used to associate with the Azure Stack HCI Logical Network. Possible switch name can be retrieved following the [Azure documents](https://learn.microsoft.com/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites). Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `virtual_switch_name` - (Required) The name of the virtual switch on the cluster used to associate with the Azure Stack HCI Logical Network. Possible switch names can be retrieved by following this [Azure guide](https://learn.microsoft.com/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites). Changing this forces a new resource to be created.
```

Author: changed

* `end` - (Required) The IPv4 address of the end of the IP address pool. Changing this forces a new resource to be created.

---

A `route` block supports the following:

* `name` - (Required) The name of the route. Changing this forces a new resource to be created.

* `address_prefix` - (Optional) The Address in CIDR notation. Changing this forces a new resource to be created.

* `next_hop_ip_address` - (Optional) The IPv4 address of the next hop. Changing this forces a new resource to be created.

---

A `subnet` block supports the following:

* `ip_allocation_method` - (Required) IP address allocation method for the subnet. Possible values are `Dynamic` and `Static`. Changing this forces a new resource to be created.

* `address_prefix` - (Optional) The address prefix in CIDR notation. Changing this forces a new resource to be created.

* `ip_pool` - (Optional) One or more `ip_pool` block as defined above. Changing this forces a new resource to be created.

* `route` - (Optional) One or more `route` block as defined above. Changing this forces a new resource to be created.

> * `vlan_id` - (Optional) VLAN ID for the Logical Network. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `vlan_id` - (Optional) The VLAN ID for the Logical Network. Changing this forces a new resource to be created.
```

Author: changed

  vm_switch_name      = "ConvergedSwitch(managementcompute)"
  dns_servers         = ["10.0.0.7", "10.0.0.8"]

  subnet {
    ip_allocation_method = "Static"
    address_prefix       = "10.0.0.0/24"
    ip_pool {
      start = "10.0.0.218"
      end   = "10.0.0.230"
    }
    route {
      name                = "example-route"
      address_prefix      = "10.0.0.0/28"
      next_hop_ip_address = "10.0.20.1"
    }
    vlan_id = 123
  }

  tags = {
    foo = "bar"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Azure Stack HCI Logical Network. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

* `custom_location_id` - (Required) The ID of Custom Location where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.

> * `vm_switch_name` - (Required) The name of the network switch used to associate with the Azure Stack HCI Logical Network. Possible switch name can be retrieved following the [Azure documents](https://learn.microsoft.com/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites). Changing this forces a new resource to be created.

Reviewer 2: from the description, it seems should be named as `network_switch_name` ?

Author: changed to `virtual_switch_name` and the description to `Name of the external virtual switch`

* `subnet` - (Required) A `subnet` block as defined below. Changing this forces a new resource to be created.

* `dns_servers` - (Optional) A list of IPv4 addresses of DNS servers available to VMs deployed in the Logical Networks. Changing this forces a new resource to be created.

* `tags` - (Optional) A mapping of tags which should be assigned to the Azure Stack HCI Logical Network.

---

A `ip_pool` block supports the following:

* `start` - (Required) The IPv4 address of the start of the IP address pool. Changing this forces a new resource to be created.

* `end` - (Required) The IPv4 address of the end of the IP address pool. Changing this forces a new resource to be created.

---

A `route` block supports the following:

* `name` - (Required) The name of the route. Changing this forces a new resource to be created.

* `address_prefix` - (Optional) The Address in CIDR notation. Changing this forces a new resource to be created.

* `next_hop_ip_address` - (Optional) The IPv4 address of the next hop. Changing this forces a new resource to be created.

---

A `subnet` block supports the following:

> * `ip_allocation_method` - (Required) IP address allocation method for the subnet. Possible value is `Dynamic` or `Static`. Changing this forces a new resource to be created.

Reviewer 2: ```suggestion
* `ip_allocation_method` - (Required) IP address allocation method for the subnet. Possible values are `Dynamic` and `Static`. Changing this forces a new resource to be created.
```

Author: fixed

* `address_prefix` - (Optional) The address prefix in CIDR notation. Changing this forces a new resource to be created.

> * `ip_pool` - (Optional) One or more `ip_pool` block as defined above. Changing this forces a new resource to be created.

Reviewer 2: why not define the description of `ip_pool` and `route` after `subnet` ?

Author: all the blocks are ranked in alphabetical order
